### PR TITLE
Add json escape option

### PR DESCRIPTION
**Pull request by g-k**
refs: https://github.com/hapijs/hoek/pull/204

Adds the `json.escape` option to call `hoek.escapeJSON` on the JSON output. It's disabled by default.

There's probably a better place and name for this option.

### CODE REVIEW

File: API.md [hunk 1]
```diff
@@ -2521,10 +2521,11 @@ following options:
   for internal routes that should not be accessible to the outside world. Defaults to `false`.
 
 - `json` - optional arguments passed to `JSON.stringify()` when converting an object or
-  error response to a string payload. Supports the following:
+  error response to a string payload or escaping it after stringification. Supports the following:
     - `replacer` - the replacer function or array. Defaults to no action.
     - `space` - number of spaces to indent nested object keys. Defaults to no indentation.
     - `suffix` - string suffix added after conversion to JSON string. Defaults to no suffix.
+    - `escape` - calls [`Hoek.jsonEscape`](https://github.com/hapijs/hoek/blob/master/API.md#escapejsonstring) after conversion to JSON string. Defaults to not escaping.
```
Comment (AdriVanHoudt) [inline API.md]: I would go with "defaults to `false`."

File: package.json [hunk 2]
```diff
@@ -26,7 +26,7 @@
     "catbox-memory": "2.x.x",
     "cryptiles": "3.x.x",
     "heavy": "4.x.x",
-    "hoek": "4.x.x",
+    "hoek": "git+https://git@github.com/hapijs/hoek.git",
```
Comment (AdriVanHoudt) [inline package.json]: This should be a legit version, maybe ask @nlf to do a release of hoek?

File: test/response.js [hunk 3]
```diff
@@ -800,6 +800,26 @@ describe('Response', () => {
         });
     });
 
+    describe('escape()', () => {
```
Comment (AdriVanHoudt) [inline test/response.js]: Add a test with a success case as well for `.escape()`

**Comment by AdriVanHoudt** [comment]
Also CI is failing since, I think, hapi uses a shrinkwrap that causes it to install hoek@4 and not the latest version you told it to in the package.json. A new version of hoek would solve this.

**Comment by hueniverse** [comment]
@g-k you still have some open comments above (beside the missing hoek version).

**Pull request merged**